2018swecapstone/h2ms
# Adding a Frontend for resetting passwords

### Please give a short summary of what's included in this submission
I'm adding a front end for resetting passwords. This includes
* A forgot password page where a user can enter their email
* A reset password page where they will be directed after they are sent a reset-password email
* Minor changes to the PasswordController to accept a callback url (I need this to forward me to the reset password page)

#### To test
1) Follow steps 1-3 from https://github.com/2018swecapstone/h2ms/wiki/Testing-password-reset-back-end-mechanism
2) Open config.service.ts, change the defaultConfig to h2msLocal.CONFIG on line 15
3) Go to the login page
4) Click forgot password
5) Fill in "admin@h2ms.org" and send email
6) Check email, click reset password link
7) Reset password
8) Login with new password

@WesleySkillern or @K-Tros Please look over my front end changes, primarily the forgot-password and reset-password pages.
@shinwachi or @stbenjam Please look over my changes to the PasswordController

### Checklist
- [ ] The Pull Request title describes the changes I've made
- [ ] I've reviewed the **Files changed** tab
- [ ] The code in the **Files changed** tab meets our style guides
- [ ] New classes and public methods in the **Files changed** tab are documented
- [ ] I've added two reviewers for this pull request
